jdaviz/configs/cubeviz/plugins/viewers.py

```diff
@@ -65,7 +65,8 @@ def on_mouse_or_key_event(self, data):
 
             # Extract first dataset from visible layers and use this for coordinates - the choice
             # of dataset shouldn't matter if the datasets are linked correctly
-            image = visible_layers[0].layer
+            active_layer = visible_layers[-1]
```

Comment:
This is actually a real fix for Imviz (see #1392), so we will need this one as a separate PR at some point when things are less confusing.